gst/nnstreamer/tensor_source/tensor_src_iio.c

```diff
@@ -1619,6 +1650,34 @@ gst_tensor_src_iio_setup_scan_channels (GstTensorSrcIIO * self)
     }
   }
 
+  /** enable the custom channels and disable the rest */
+  if (self->channels_enabled == CHANNELS_ENABLED_CUSTOM) {
+    for (ch_list = self->channels; ch_list != NULL; ch_list = ch_list->next) {
+      channel_prop = (GstTensorSrcIIOChannelProperties *) ch_list->data;
+      item =
+          g_list_find (self->custom_channel_list,
```

Comment:
Could it be possible to use more efficient mechanism?
Searching for a linked list for every ch_list appears to be less efficient.

If the number of channels is limited to 32 or 64, you may simply use bitmap with uint32 or uint64, simplifying ```_GstTensorSrcIIO``` as well.

Reply:
I cannot find any upper limit on the the number of channels for a device.

What do you think about using GArray instead of GList, sort the array (`custom_channel_list`) and then use binary search (binary search will have to be implemented).

Reply:
https://developer.gnome.org/glib/stable/glib-Hash-Tables.html may be fast enough. You may use this as a key-value storage without values.

Reply:
updated using hashtable.